app/(gcforms)/[locale]/(app administration)/admin/(with nav)/settings/actions.ts

```diff
@@ -4,51 +4,65 @@ import { createAbility } from "@lib/privileges";
 import {
   createAppSetting,
   deleteAppSetting,
-  getAppSetting,
+  getFullAppSetting,
   updateAppSetting,
 } from "@lib/appSettings";
 import { revalidatePath } from "next/cache";
 
-// Note: any thrown errors will be caught in the Error boundary/component
-
-export async function getSetting(internalId: string) {
+const authCheck = async () => {
   const session = await auth();
-  if (!session) throw new Error("No session");
+  if (!session) throw new Error("No session found");
+  return createAbility(session);
+};
 
-  await getAppSetting(internalId);
-  revalidatePath("(gcforms)/[locale]/(app administration)/admin/(with nav)/settings");
+export async function getSetting(internalId: string) {
+  await authCheck();
+  return getFullAppSetting(internalId);
 }
 
-export async function updateSetting(internalId: string, setting: Record<string, unknown>) {
-  const session = await auth();
-  if (!session) throw new Error("No session");
-  const ability = createAbility(session);
-
-  await updateAppSetting(ability, internalId, setting);
+export async function updateSetting(formData: FormData) {
+  const ability = await authCheck();
+  const setting = {
+    internalId: formData.get("internalId") as string,
+    nameEn: formData.get("nameEn") as string,
+    nameFr: formData.get("nameFr") as string,
+    descriptionEn: formData.get("descriptionEn") as string,
+    descriptionFr: formData.get("descriptionFr") as string,
+    value: formData.get("value") as string,
+  };
+  await updateAppSetting(ability, setting.internalId, setting).catch(() => {
+    throw new Error("Error updating setting");
+  });
   revalidatePath("(gcforms)/[locale]/(app administration)/admin/(with nav)/settings");
 }
 
-export async function createSetting(internalId: string, setting: Record<string, unknown>) {
-  const session = await auth();
-  if (!session) throw new Error("No session");
-  const ability = createAbility(session);
-
+export async function createSetting(formData: FormData) {
+  const ability = await authCheck();
+  const setting = {
+    internalId: formData.get("internalId") as string,
+    nameEn: formData.get("nameEn") as string,
+    nameFr: formData.get("nameFr") as string,
+    descriptionEn: formData.get("descriptionEn") as string,
+    descriptionFr: formData.get("descriptionFr") as string,
+    value: formData.get("value") as string,
+  };
   await createAppSetting(
     ability,
     setting as {
       internalId: string;
       nameEn: string;
       nameFr: string;
     }
-  );
+  ).catch(() => {
+    throw new Error("Error creating setting");
+  });
   revalidatePath("(gcforms)/[locale]/(app administration)/admin/(with nav)/settings");
```

Comment:
`revalidatePath` requires the "page" or "layout" argument when a dynamic parameter is included in the path.
In this case [locale].